src/components/forms/CreateNhsAppTemplate/CreateNhsAppTemplate.tsx

```diff
@@ -49,44 +45,31 @@ export const CreateNhsAppTemplate: FC<PageComponentProps> = ({
     router.push(state.redirect);
   }
 
-  const [templateName, setTemplateName] = useState(state.nhsAppTemplateName);
-  const [templateMessage, setTemplateMessage] = useState(
+  const [nhsAppTemplateMessage, nhsAppMessageHandler] = useTextAreaInput(
     state.nhsAppTemplateMessage
   );
-  const [jsEnabledStyle, setJsEnabledStyle] = useState<
-    CSSProperties | undefined
-  >({ display: 'none' });
-
-  useEffect(() => {
-    setJsEnabledStyle(undefined);
-  }, []);
-
-  const templateNameHandler: FormEventHandler<HTMLInputElement> = (event) => {
-    const typedEventTarget = event.target as HTMLInputElement; // it would be great if we could do this without forcing the types
-    setTemplateName(typedEventTarget.value);
-  };
-
-  const templateMessageHandler: FormEventHandler<HTMLTextAreaElement> = (
-    event
-  ) => {
-    const typedEventTarget = event.target as HTMLTextAreaElement; // it would be great if we could do this without forcing the types
-    setTemplateMessage(typedEventTarget.value);
-  };
 
   const templateNameError =
     state.validationError?.fieldErrors.nhsAppTemplateName?.join(', ');
 
+  const templateMessageError =
+    state.validationError?.fieldErrors.nhsAppTemplateMessage?.join(', ');
+
   return (
     <div className='nhsuk-grid-row'>
       <NHSNotifyBackButton
         formId='create-nhs-app-template-back'
         action={action}
       >
-        <input type='hidden' name='nhsAppTemplateName' value={templateName} />
+        <input
+          type='hidden'
+          name='nhsAppTemplateName'
+          value={state.nhsAppTemplateName}
+        />
         <input
           type='hidden'
           name='nhsAppTemplateMessage'
-          value={templateMessage}
+          value={state.nhsAppTemplateMessage}
```

Comment:
why aren't we tracking changes to the template name anymore?

Reply:
Ah yes! I've updated it now and replicated the changes on SMS Create.